brain/synapse.h

```diff
@@ -0,0 +1,138 @@
+/* Copyright (c) 2013-2016, EPFL/Blue Brain Project
+ *                          Daniel.Nachbaur@epfl.ch
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef BRAIN_SYNAPSE
+#define BRAIN_SYNAPSE
+
+#include <brain/api.h>
+#include <brain/types.h>
+
+#include <memory>
+
+namespace brain
+{
+
+/**
+ * A proxy object returned by the Synapses container to access data for a
+ * particular synapse.
+ *
+ * The lifetime of this object is stricly bound to the synapses container it
+ * comes from.
+ */
+class Synapse
+{
+public:
+    /** Copy construct this synapse by sharing the underlying data. */
+    BRAIN_API Synapse( const Synapse& );
+
+    /** Move construct this synapse. */
+    BRAIN_API Synapse( Synapse&& rhs );
+
+    /** Move assign this synapse. */
+    BRAIN_API Synapse& operator=( Synapse&& rhs );
+
+    BRAIN_API ~Synapse();
+
```

Comment:
What about an operator==. Note that comparisons between synapses from different containers are reasonable as long as they come from the same circuit.